File: src/app/_models/chatting/chatDto.ts
import { ChatUserDto } from "./chatUserDto"
import { MessageDto } from "./messageDto"

export interface ChatDto {
    id: string
    participants: ChatUserDto[]
    lastMessage: string
    unreadCount: number
}

export interface ChatWithMessageDto {
    id: string,
    participants: ChatUserDto[],
    lastMessage: string,
    unreadCount: number,
    messages: MessageDto[]
}

File: src/Domain/DTOs/MessagingDTOs/ChatDto.cs
namespace Domain.DTOs.MessagingDTOs;

public class ChatDto
{
    public string Id { get; set; } = string.Empty;
    public List<ChatUserDto> Participants { get; set; } = null!;
    public string LastMessage { get; set; } = string.Empty;
    public int UnreadCount { get; set; }

}